File: Codeforces/_1831_D.java
import java.util.*;
import java.io.*;

public class _1831_D {
    public static void main(String[] args) throws IOException {
        Reader in = new Reader();
        PrintWriter out = new PrintWriter(System.out);
        int t = in.nextInt();

        int[][] count = new int[635][200005];
        while (t-- > 0) {
            int n = in.nextInt();
            int[] a = new int[n];
            for (int i = 0; i < n; i++) {
                a[i] = in.nextInt();
            }
            int[] b = new int[n];
            for (int i = 0; i < n; i++) {
                b[i] = in.nextInt();
            }

            int sqrt = 1;
            while (sqrt * sqrt <= 2 * n) {
                sqrt++;
            }
            sqrt--;

            // count[x][y] counts the number of indices i with a_i = x and b_i = y, only x
            // up to sqrt(n)
            for (int i = 0; i < n; i++) {
                if (a[i] <= sqrt) {
                    count[a[i]][b[i]]++;
                }
            }

            long count1 = 0;
            long count2 = 0;
            for (int i = 0; i < n; i++) {
                for (int j = 1; j <= sqrt; j++) {
                    int prod = a[i] * j;
                    if (prod > 2 * n) {
                        break;
                    }
                    if (prod - b[i] > 0 && prod - b[i] <= n) {
                        if (j == a[i] && prod == b[i] * 2) {
                            count1 += Math.max(count[j][prod - b[i]] - 1, 0);
                        } else if (a[i] <= sqrt) {
                            count1 += count[j][prod - b[i]];
                        } else {
                            count2 += count[j][prod - b[i]];
                        }
                    }
                }
            }
            out.println(count1 / 2 + count2);

            for (int i = 0; i < n; i++) {
                if (a[i] <= sqrt) {
                    count[a[i]][b[i]]--;
                }
            }
        }
        in.close();
        out.close();
    }

    static class Reader {
        final private int BUFFER_SIZE = 1 << 16;
        private DataInputStream din;
        private byte[] buffer;
        private int bufferPointer, bytesRead;

        public Reader() {
            din = new DataInputStream(System.in);
            buffer = new byte[BUFFER_SIZE];
            bufferPointer = bytesRead = 0;
        }

        public Reader(String file_name) throws IOException {
            din = new DataInputStream(new FileInputStream(file_name));
            buffer = new byte[BUFFER_SIZE];
            bufferPointer = bytesRead = 0;
        }

        public String readLine() throws IOException {
            byte[] buf = new byte[64]; // line length
            int cnt = 0, c;
            while ((c = read()) != -1) {
                if (c == '\n') {
                    if (cnt != 0) {
                        break;
                    } else {
                        continue;
                    }
                }
                buf[cnt++] = (byte) c;
            }
            return new String(buf, 0, cnt);
        }

        public int nextInt() throws IOException {
            int ret = 0;
            byte c = read();
            while (c <= ' ') {
                c = read();
            }
            boolean neg = (c == '-');
            if (neg) {
                c = read();
            }
            do {
                ret = ret * 10 + c - '0';
            } while ((c = read()) >= '0' && c <= '9');
            if (neg) {
                return -ret;
            }
            return ret;
        }

        public long nextLong() throws IOException {
            long ret = 0;
            byte c = read();
            while (c <= ' ') {
                c = read();
            }
            boolean neg = (c == '-');
            if (neg) {
                c = read();
            }
            do {
                ret = ret * 10 + c - '0';
            } while ((c = read()) >= '0' && c <= '9');
            if (neg) {
                return -ret;
            }
            return ret;
        }

        public double nextDouble() throws IOException {
            double ret = 0, div = 1;
            byte c = read();
            while (c <= ' ') {
                c = read();
            }
            boolean neg = (c == '-');
            if (neg) {
                c = read();
            }
            do {
                ret = ret * 10 + c - '0';
            } while ((c = read()) >= '0' && c <= '9');
            if (c == '.') {
                while ((c = read()) >= '0' && c <= '9') {
                    ret += (c - '0') / (div *= 10);
                }
            }
            if (neg) {
                return -ret;
            }
            return ret;
        }

        private void fillBuffer() throws IOException {
            bytesRead = din.read(buffer, bufferPointer = 0, BUFFER_SIZE);
            if (bytesRead == -1) {
                buffer[0] = -1;
            }
        }

        private byte read() throws IOException {
            if (bufferPointer == bytesRead) {
                fillBuffer();
            }
            return buffer[bufferPointer++];
        }

        public void close() throws IOException {
            if (din == null) {
                return;
            }
            din.close();
        }
    }
}

File: Codeforces/_1831_D.cpp
#include <bits/stdc++.h>
using namespace std;

int count_arr[635][200005] = {0};
void solve()
{
    int n;
    cin >> n;
    int a[n] = {0};
    int b[n] = {0};
    for (int i = 0; i < n; i++)
    {
        cin >> a[i];
    }
    for (int i = 0; i < n; i++)
    {
        cin >> b[i];
    }

    int fl_sq = 1;
    while (fl_sq * fl_sq <= 2 * n)
    {
        fl_sq++;
    }
    fl_sq--;

    // count_arr[x][y] counts the number of indices i with a_i = x and b_i = y, only x up to sqrt(n)
    for (int i = 0; i < n; i++)
    {
        if (a[i] <= fl_sq)
        {
            count_arr[a[i]][b[i]]++;
        }
    }

    long long count1 = 0;
    long long count2 = 0;
    for (int i = 0; i < n; i++)
    {
        for (int j = 1; j <= fl_sq; j++)
        {
            int prod = a[i] * j;
            if (prod > 2 * n)
            {
                break;
            }
            if (prod - b[i] > 0 && prod - b[i] <= n)
            {
                if (j == a[i] && prod == b[i] * 2)
                {
                    count1 += max(count_arr[j][prod - b[i]] - 1, 0);
                }
                else if (a[i] <= fl_sq)
                {
                    count1 += count_arr[j][prod - b[i]];
                }
                else
                {
                    count2 += count_arr[j][prod - b[i]];
                }
            }
        }
    }
    long long res = count1 / 2 + count2;
    cout << res << "\n";

    for (int i = 0; i < n; i++)
    {
        if (a[i] <= fl_sq)
        {
            count_arr[a[i]][b[i]]--;
        }
    }
}

int main()
{
    int t;
    cin >> t;
    while (t--)
    {
        solve();
    }
}